packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1574,6 +1574,31 @@ void main() {
     expect(RawKeyboard.instance.keyEventHandler, same(rawKeyEventHandler));
   });
 
+  testWidgets('Incorrect use of ParentDataWidget shows a useful error message', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      SizedBox(
+        key: const Key('container'),
+        child: Flexible( // Flexible should not be used directly in a Container or other inappropriate parents
```

Comment:
If you want to place a comment here, you can put it above the Flexible. That makes the formatting a bit nicer.
```suggestion
        // Flexible should only be used inside Flex widgets, such as a Column or Row.
        child: Flexible(
```